sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -52,29 +52,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
    * Returns true when all operators are integral.
    */
   private def checkSpecialExpressionIntegrity(plan: LogicalPlan): Boolean = {
```

Comment:
It seems this private function is called by `isPlanIntegral` only, so we need to define this separately?

Reply:
@maropu Ok... Let me remove it.